R/comment_linters.R

```diff
@@ -54,7 +53,10 @@ commented_code_linter <- function() {
     all_comment_nodes <- xml2::xml_find_all(source_expression$full_xml_parsed_content, "//COMMENT")
     all_comments <- xml2::xml_text(all_comment_nodes)
     code_candidates <- re_matches(all_comments, code_candidate_regex, global = FALSE, locations = TRUE)
-    is_parsable <- which(vapply(code_candidates[, "code"], parsable, logical(1L)))
+    extracted_code <- code_candidates[, "code"]
+    # ignore trailing ',' when testing for parsability
+    extracted_code <- rex::re_substitutes(extracted_code, rex::rex(",", any_spaces, end), "")
```

Comment:
include trailing comments too?

```r
# na.rm = TRUE, # ignore missing entries
```

Reply:
That's not so easy to properly detect without introducing false negatives I think: `x <- "this#should lint"`.

Or am I missing something?

Reply:
Hmm I had in mind `,\s*(#.*)?$`, but you're right it technically wouldn't lint on `# "na.rm = TRUE, # comment in string"`. Seems a bit obscure, but we can consider that later if a user requests.